### PR TITLE
Segregation of role Done

### PR DESCRIPTION
**Pull request by sverma58591**
I have segregated the role of Student and teacher after sign in.

### CODE REVIEW

File: EMentors/app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,12 +2,19 @@ class ApplicationController < ActionController::Base
     before_action :configure_permitted_parameters, if: :devise_controller?
     protected
     def after_sign_in_path_for(resource)
-        home_index_path
+        if current_user.role == 'teacher'
```
Comment (rahulgrepruby) [inline EMentors/app/controllers/application_controller.rb]: please add a method **is_teacher? or is_student?** and directly call 
`resource.is_teacher? ? teachers_index_path : students_index_path`

File: EMentors/app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -51,9 +51,12 @@ class Users::RegistrationsController < Devise::RegistrationsController
   # end
 
   # The path used after sign up.
-  # def after_sign_up_path_for(resource)
-  #   super(resource)
-  # end
+  def after_sign_up_path_for(resource)
+    # super(resource)
+    # byebug
+    sign_out resource
```
Comment (rahulgrepruby) [inline EMentors/app/controllers/users/registrations_controller.rb]: override registration#create method for this

File: EMentors/app/views/users/registrations/new.html.erb [hunk 3]
```diff
@@ -50,7 +50,11 @@
                   <%# <input type="email" id="form3Example3" class="form-control" />
                   <label class="form-label" for="form3Example3">Email address</label> %>
                 </div>
-                
+                <div class="form-outline mb-4">
+                    <%= f.select :role, collection: User.roles.keys.to_a %>
```
Comment (rahulgrepruby) [inline EMentors/app/views/users/registrations/new.html.erb]: .keys method will returns array remove to_a

**Review by rahulgrepruby — APPROVED** [review]
consider my comments and merge it

File: EMentors/app/controllers/users/registrations_controller.rb (hunk 2 above)
Comment (sverma58591) [reply EMentors/app/controllers/users/registrations_controller.rb]: Sir I have added confirmable to the devise and now it is not creating sessions. Like after sign up action it is redirecting to the sign in page and also flashing the message your email is confirmed.

File: EMentors/app/views/users/registrations/new.html.erb (hunk 3 above)
Comment (sverma58591) [reply EMentors/app/views/users/registrations/new.html.erb]: Done sir.

**Pull request merged**